Package.swift

```diff
@@ -3,183 +3,188 @@
-    let featureSet = ExperimentalFeature.allCases.filter { env.keys.contains($0.environmentFlag!) }
-    return ExperimentalFeature(featureSet)
+func selectCombineImp() -> CombineImplementation {
+    let imp = env["CX_IMPLEMENTATION"]?.lowercased()
+    switch imp {
+    case "combine":         return .combine
+    case "combinex":        return .combineX
+    case "opencombine":     return .openCombine
+    default:
+        #if canImport(Combine)
+        return .combine
+        #else
+        return .combineX
+        #endif
+    }
 }
 
-let combineImpl: CombineImplementation = selectCombineImpl()
-let experimentalFeatures: ExperimentalFeature = configExperimentalFeatures()
-
-let testSwiftSettings: [SwiftSetting] = [.define(combineImpl.swiftFlag)] + (experimentalFeatures.swiftSettings ?? [])
-
-// MARK: - Package
-
-let package = Package(
-    name: "CombineX",
-    platforms: [
-        .macOS(.v10_10),
-        .iOS(.v8),
-        .tvOS(.v9),
-        .watchOS(.v2),
-    ],
-    products: [
-        .library(
-            name: "CombineX",
-            targets: ["CombineX", "CXFoundation"]),
-        .library(
-            name: "CXCompatible",
-            targets: ["CXCompatible"]),
-        .library(
-            name: "CXShim",
-            targets: ["CXShim"]),
-    ],
-    dependencies: [
-        .package(url: "https://github.com/Quick/Quick.git", from: "2.0.0"),
-        .package(url: "https://github.com/Quick/Nimble.git", from: "8.0.0"),
-    ] + combineImpl.packageDependencies + experimentalFeatures.packageDependencies,
-    targets: [
-        .target(
-            name: "cxlibc"),
-        .target(
-            name: "CXUtility"),
-        .target(
-            name: "CXNamespace"),
-        .target(
-            name: "CombineX",
-            dependencies: ["cxlibc", "CXUtility", "CXNamespace"] + experimentalFeatures.targetDependencies,
-            swiftSettings: experimentalFeatures.swiftSettings),
-        .target(
-            name: "CXFoundation",
-            dependencies: ["CXUtility", "CXNamespace", "CombineX"]),
-        .target(
-            name: "CXCompatible",
-            dependencies: ["CXNamespace"]),
-        .target(
-            name: "CXShim",
-            dependencies: combineImpl.targetDependencies,
-            swiftSettings: [.define(combineImpl.swiftFlag)]),
-        
-        // MARK: Tests
-        .testTarget(
-            name: "CombineXTests",
-            dependencies: ["CXUtility", "CXShim", "Quick", "Nimble"],
-            swiftSettings: testSwiftSettings),
-        .testTarget(
-            name: "CXFoundationTests",
-            dependencies: ["CXShim", "Quick", "Nimble"],
-            swiftSettings: testSwiftSettings),
-    ],
-    swiftLanguageVersions: [
-        .v5
-    ]
-)
+let currentCombineImp = selectCombineImp()
+currentCombineImp.configure(package)
 
+// Pass the swift settings of current combine implementation and all experimental features to all test targets.
```

Comment:
I think experimental features should affect test target as well. These test should pass normally or fail as expected.